app.rb

```diff
@@ -0,0 +1,169 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rental'
+require_relative 'book'
+require_relative 'classroom'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+    # @classrooms = Classroom.new('Form 5')
+  end
+
+  def menu
+    puts
+    puts 'Please choose an option by entering a number'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+  end
+
+  def run
+    puts 'Welcome to school library'
+    loop do
+      menu
+      option = gets.chomp
+      break if option == '7'
+
+      get_option(option)
+    end
+    puts 'Thank you for using our library!'
+  end
+
+  def get_option(option)
+    case option
+    when '1'
+      list_all_books
+    when '2'
+      list_all_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_rentals_by_person_id
+    else
+      puts 'Thank you for using this App!'
+    end
+  end
+
+  def list_all_books
+    puts 'There are no books yet!.' if @books.empty?
+
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+    sleep 0.75
+  end
+
+  def list_all_people
+    puts 'There is not a person yet!.' if @people.empty?
+    @people.each do |person|
+      puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    sleep 0.75
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or teacher (2) [Input a number]: '
+    person_option = gets.chomp
+
+    case person_option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Incorrect input please use 1 or 2'
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
+
+    student = Student.new(@classrooms, age, name, parent_permission)
+    @people << student
+
+    puts 'Person created successfully!'
+    sleep 0.75
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    teacher = Teacher.new(specialization, age, name)
+    @people << teacher
+
+    puts 'Teacher created successfully'
+    sleep 0.75
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+
+    print 'Author: '
+    author = gets.chomp
+
+    book = Book.new(title, author)
+    @books << book
+
+    puts 'Book added successfully'
+    sleep 0.75
+  end
+end
+
+def create_rental
```

Comment:
- [ ] Your app is breaking in attempt to create a rental for a person. Just after a date is entered for rental the app crashes. Kindly ensure to improve upon this aspect.

**Creating rental error:**
![image](https://user-images.githubusercontent.com/12745474/181076477-54320885-26d2-498b-bb15-1e18fbee4950.png)